lib/task/migrate/migrations/arMigration0177.class.php

```diff
@@ -75,102 +75,27 @@ public function up($configuration)
         'column' => 'object_id',
         'index' => 'job_FI_3',
       ),
-    );
-
-    foreach ($indexes as $data)
-    {
-      // Get actual index name
-      $sql = 'SHOW INDEX FROM %s WHERE Column_name=:column_name;';
-      $result = QubitPdo::fetchOne(
-        sprintf($sql, $data['table']),
-        array(':column_name' => $data['column'])
-      );
-
-      // Stop if the index is missing
-      if (!$result || !$result->Key_name)
-      {
-        throw new Exception(sprintf(
-          "Could not find index for '%s' column on '%s' table.",
-          $data['column'],
-          $data['table']
-        ));
-      }
-
-      // Skip if the index already has the expected name
-      if ($result->Key_name == $data['index'])
-      {
-        continue;
-      }
-
-      QubitPdo::modify(sprintf(
-        'ALTER TABLE %s RENAME INDEX %s TO %s;',
-        $data['table'],
-        $result->Key_name,
-        $data['index']
-      ));
-    }
+    ));
 
     // Fix foreign keys:
     // - Restore ON DELETE CASCADE on DO `object_id` (missing on migration 169).
-    // - Rename IO `display_standard_id` constraint (unnamed on migration 94)
-    $foreignKeys = array(
+    // - Rename IO `display_standard_id` constraint (unnamed on migration 94).
+    QubitMigrate::updateForeignKeys(array(
       array(
-        'tableName' => 'digital_object',
-        'columnName' => 'object_id',
-        'refTableName' => 'object',
-        'newConstraintName' => 'digital_object_FK_2',
-        'onDelete' => 'ON DELETE CASCADE',
+        'table' => 'digital_object',
+        'column' => 'object_id',
+        'refTable' => 'object',
+        'constraint' => 'digital_object_FK_2',
+        'onDelete' => ' ON DELETE CASCADE',
```

Comment:
I'd prefer to see the leading space in the SQL templates, rather than having it in the 'onDelete' value.  To me it seems too easy to forgot to add the leading space when adding new migrations.

Reply:
Great point!